src/NodeApi/JSObject.cs

```diff
@@ -9,11 +9,16 @@
 namespace Microsoft.JavaScript.NodeApi;
 
 public readonly partial struct JSObject : IDictionary<JSValue, JSValue>, IEquatable<JSValue>
+#if NET7_0_OR_GREATER
+    , IJSValue<JSObject>
+#endif
 {
     private readonly JSValue _value;
 
-    public static explicit operator JSObject(JSValue value) => new(value);
-    public static implicit operator JSValue(JSObject obj) => obj._value;
+    public static implicit operator JSValue(JSObject value) => value.AsJSValue();
+    public static explicit operator JSObject?(JSValue value) => value.As<JSMap>();
+    public static explicit operator JSObject(JSValue value)
+        => value.As<JSObject>() ?? throw new InvalidCastException("JSValue is not an Object.");
```

Comment:
Should there be a generic method that does the equivalent of this checked cast? How about `JSValue.CastTo<TValue>()`?

(Note we have non-generic "CoerceTo" methods already.)

Reply:
Added `JSValue.CastTo<T>`